### PR TITLE
fix(client): disable enumerable on JsonFilter path

### LINKED ISSUE #7136: Disallow passing a string as a path in Postgres
### Bug description

From @Weakky 

> One of the reason why I think we shouldn’t allow a string to be passed is that it can be misinterpreted as a support for JSON path. The reason postgres does not accept a string is because it does NOT support JSON path.
With Postgres, if you want to target a “deep” property, you need to do: path: ["a", "b"]. The equivalent on MySQL is "path": "$.a.b" (because MySQL does support JSON path) 

### How to reproduce

Follow the example in the release notes for 2.23.0 and then write

```js
const posts = await prisma.post.findMany({
    where: {
      content: {
        path: "type",
      },
    },
  })
```

### Expected behavior

- Postgres: `path: string[]`
- MySQL: `path: string`

### Prisma information

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["filterJson"]
}

model Log {
  id      Int    @id @default(autoincrement())
  level   Level
  message String
  meta    Json
}

enum Level {
  INFO
  WARN
  ERROR
}

```

### Environment & setup

- OS: OSX
- Database: Postgres

### Prisma Version

```
prisma               : 2.23.0-dev.41
@prisma/client       : 2.23.0-dev.41
Current platform     : darwin
Query Engine         : query-engine adf5e8cba3daf12d456d911d72b6e9418681b28b (at node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli adf5e8cba3daf12d456d911d72b6e9418681b28b (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core adf5e8cba3daf12d456d911d72b6e9418681b28b (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt adf5e8cba3daf12d456d911d72b6e9418681b28b (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : adf5e8cba3daf12d456d911d72b6e9418681b28b
Studio               : 0.393.0
Preview Features     : filterJson
```

### PR DESCRIPTION
**Pull request by williamluke4**
implements https://github.com/prisma/prisma/issues/7136

### CODE REVIEW

File: src/packages/client/src/__tests__/types/json-filtering-mysql/index.test-d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { PrismaClient } from '.'
+import { expectError } from 'tsd'
+
+// tslint:disable
```
Comment (Jolg42) [inline src/packages/client/src/__tests__/types/json-filtering-mysql/index.test-d.ts:4]: Why is that needed? 👀

Comment (williamluke4) [reply src/packages/client/src/__tests__/types/json-filtering-mysql/index.test-d.ts]: No idea was copy/pasted from another test

File: src/packages/client/src/__tests__/types/json-filtering-mysql/test.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { PrismaClient, Prisma } from '@prisma/client'
+
+// tslint:disable
```
Comment (Jolg42) [inline src/packages/client/src/__tests__/types/json-filtering-mysql/test.ts:3]: One more here

File: src/packages/client/src/__tests__/types/json-filtering-postgres/test.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { PrismaClient, Prisma } from '@prisma/client'
+
+// tslint:disable
```
Comment (Jolg42) [inline src/packages/client/src/__tests__/types/json-filtering-postgres/test.ts:3]: and here :)

**Review by Jolg42 — APPROVED** [review]
Looks good to me 💯

I think @millsp could also do a quick review

**Review by millsp — APPROVED** [review]
